backend/dataall/modules/dataset_sharing/services/data_sharing_service.py

```diff
@@ -76,85 +76,107 @@ def approve_share(cls, engine: Engine, share_uri: str) -> bool:
+                                session,
+                                share_uri,
+                                folder.locationUri,
+                            )
+                            cls.handle_share_items_failure_during_locking(
+                                session, share_item, ShareItemStatus.Share_Approved.value
+                            )
+
+                        for bucket in shared_buckets:
+                            share_item = ShareObjectRepository.find_sharable_item(
+                                session,
+                                share_uri,
+                                bucket.bucketUri,
+                            )
+                            cls.handle_share_items_failure_during_locking(
+                                session, share_item, ShareItemStatus.Share_Approved.value
+                            )
+
+                        share_object_SM = ShareObjectSM(share.status)
+                        new_object_state = share_object_SM.run_transition(ShareObjectActions.AcquireLockFailure.value)
+                        share_object_SM.update_state(session, share, new_object_state)
+                        return False
+
+                    log.info(f'Granting permissions to folders: {shared_folders}')
+
+                    approved_folders_succeed = ProcessS3AccessPointShare.process_approved_shares(
+                        session,
+                        dataset,
+                        share,
+                        shared_folders,
+                        source_environment,
+                        target_environment,
+                        source_env_group,
+                        env_group,
+                    )
+                    log.info(f'sharing folders succeeded = {approved_folders_succeed}')
+
+                    log.info('Granting permissions to S3 buckets')
+
+                    approved_s3_buckets_succeed = ProcessS3BucketShare.process_approved_shares(
+                        session,
+                        dataset,
+                        share,
+                        shared_buckets,
+                        source_environment,
+                        target_environment,
+                        source_env_group,
+                        env_group,
+                    )
+                    log.info(f'sharing s3 buckets succeeded = {approved_s3_buckets_succeed}')
+
+                    log.info(f'Granting permissions to tables: {shared_tables}')
+                    approved_tables_succeed = ProcessLakeFormationShare(
+                        session,
+                        dataset,
+                        share,
+                        shared_tables,
+                        source_environment,
+                        target_environment,
+                        env_group,
+                    ).process_approved_shares()
+                    log.info(f'sharing tables succeeded = {approved_tables_succeed}')
+
```

Comment:
Very minor: principal_healthy is a weird variable name to represent the success of the shares